programs/survival/bee_healing.sc

```diff
@@ -0,0 +1,40 @@
+// Bees have an annoying habit of drowning themselves, so yeah
+__config()->{'scope' -> 'global'};
+
+global_incrementer = 0;
+global_bees = {};
+
```

Comment:
Why need a map and incrementer? If you're going to iterate over them all anyways, can't you just use a list? Also couldn't you use an entity event which schedules 800 ticks later?

Reply:
If bees don't die/unload in the reverse order they were loaded, then you would have to delete items from the middle of  the list, which would change the indexes of all the other bees in the list, which would break everything. That's why I have to use a map

I suppose you could call a recursively scheduling function for every bee that spawns, but I don't see how that would be a better implementation

Reply:
Well the recursively calling function would run for each bee, allowing for cleaner code, as you don't need to keep track of each bee, just use the entity_load_handler to assign a function like `track_health(bee)` to each bee, which incrementes the health of the bee it it's less than 10, and then at the end does `if(!bee~'removed', schedule(800, 'track_health', bee))` to call itself again if the bee is alive. In fact, that call should be put at the beginning of the function, so the whole thing looks like this:
`bee_healing.sc`:
```
__config()->{'scope'->'global'};

track_health(bee)-> if(!bee~'removed',
    health = bee~'health';
    if(health < 10, modify(bee, 'health', health+1));
    schedule(800, 'track_health', bee)
);

entity_load_handler('bee', 'track_health');
```
That would work without any bee tracking shenanigans and no iterations etc. It just assigns the event, and then leaves the bees be. (pun intended, I have now descended to the level of Cory Scheviak in terms of horrible bee puns...)

Reply:
@Xendergo I would recommend replacing bee_healing.sc with what I've posted above, as it does the exact same job without iterating over all the bees each time, instead letting each bee keep track of itself.

Reply:
Ok, I've replaced it, that definitely looks like a nicer solution